internal/controllers/s3userclaim/provisioner.go

```diff
@@ -131,6 +133,81 @@ func (r *Reconciler) ensureReadonlySubuser(ctx context.Context) (*ctrl.Result, e
 	return subreconciler.ContinueReconciling()
 }
 
+func (r *Reconciler) ensureOtherSubusers(ctx context.Context) (*ctrl.Result, error) {
+	specSubUsers := r.s3UserClaim.Spec.SubUsers
+	// Create a hashmap to move all spec and ceph subUsers to it
+	subUserFullIdSet := make(map[string]string)
+
+	// Tag specSubUsers with "create"
+	for _, subUser := range specSubUsers {
+		cephSubUserFullId := cephSubUserFullIdMaker(r.cephUserFullId, subUser)
+		subUserFullIdSet[cephSubUserFullId] = consts.SubUserTagCreate
+	}
+
+	// Add read-only subUser to subUsers to prevent if from removing
+	subUserFullIdSet[r.readonlyCephUserFullId] = consts.SubUserTagCreate
+
+	// Tag cephSubUsers as remove if they are not already in the hashmap and remove them otherwise
+	// since they are already available on ceph and not needed to created.
+	for _, cephsubUser := range r.cephUser.Subusers {
+		_, exists := subUserFullIdSet[cephsubUser.Name]
+		if exists {
+			delete(subUserFullIdSet, cephsubUser.Name)
```

Comment:
Just asking...
My understanding of the code is that we don't support direct updates, but someone can remove and recreate a sub-user if they want to update it. Is it right?

Reply:
Yes it is.